src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,104 @@
+import { SyntheticEvent, forwardRef, useRef } from "react";
+import type { AriaToastProps } from "@react-aria/toast";
+import type { ToastState, QueuedToast } from "@react-stately/toast";
+import { useToast } from "@react-aria/toast";
+import type { CustomToastContent } from "./ToastProvider";
+
+import Button from "../Button/Button";
+import { st, classes } from "./toast.st.css";
+
+interface ToastProps<T> extends AriaToastProps<T> {
+  className?: string;
+  state: ToastState<T>;
+  toast: QueuedToast<T>;
+  icon: React.ReactNode;
+}
+
+function Toast({ state, ...props }: ToastProps<CustomToastContent>) {
```

Comment:
We are using `forwardRef` (which is good because as a lib we want to always `forwardRef` where we can) but as we are not passing it in we get a console warning and ref as a prop will not be working. 
<img width="878" alt="Screenshot 2023-05-26 at 10 58 29" src="https://github.com/action-is-hope/shelley/assets/678624/65ed1ad0-493c-4543-b856-5bc7978d9177">

Tabs has the same issue so that is likely where the issue came from for you as the last comp added. Take a look at `ActionButton` as example where we use a localRef as the one for the hook and the one from props is merged using an adobe util.